### ISSUE TITLE
all those apis with pagination parameters return 429?  only on Osmosis.

### ISSUE THREAD

**Issue by liangping** [osmosis-labs/osmosis#1425]
We don't have access limit
https://osmo.api.ping.pub/cosmos/slashing/v1beta1/signing_infos?pagination.limit=5
https://lcd-osmosis.blockapsis.com/cosmos/gov/v1beta1/proposals?pagination.limit=5&pagination.reverse=true&pagination.key=

Both return this error.
```json
{
  "code": 8,
  "message": "trying to send message larger than max (6 vs. 0)",
  "details": [
  ]
}
```

**Comment by p0mvn** [comment]
Hi @liangping 

It seems to be related to new config values. Please update the following in `app.toml`:
```
max-recv-msg-size = "10485760"

max-send-msg-size = "2147483647"
```

These values are in bytes. Please feel free to adjust them according to your needs. Let me know if the problem persists

**Comment by liangping** [comment]
Thank. Let me try it

**Comment by liangping** [comment]
tried in [rpc] and [api] , but not works
```
[grpc]

# Enable defines if the gRPC server should be enabled.
enable = true

# Address defines the gRPC server address to bind to.
address = "0.0.0.0:9090"

max-recv-msg-size = "10485760"

max-send-msg-size = "2147483647"
```

**Comment by ChihuahuaChain** [comment]
> @p0mvn tried in [rpc] and [api] , but not works
> 
> ```
> [grpc]
> 
> # Enable defines if the gRPC server should be enabled.
> enable = true
> 
> # Address defines the gRPC server address to bind to.
> address = "0.0.0.0:9090"
> 
> max-recv-msg-size = "10485760"
> 
> max-send-msg-size = "2147483647"
> ```

Tried that as well and tried to increase bytes as well, won't fix it.

**Comment by ChihuahuaChain** [comment]
> @liangping @ChihuahuaChain Just FYI, I had several folks who were having this problem getting it fixed by updating these configs. Please double-check that the configs are picked up correctly. Let me know how it goes.
> 
> I will do some testing on my side today as well

Checked config, values are in, service restarted and still have these problems, I started having issues only after updating to 7.3.0, no problems prior.

**Comment by p0mvn** [comment]
I see, thanks. I'm setting up a node right now to test on my side to investigate. Let me get back to you. If anyone finds a way to resolve this in the meantime, please let me know

**Comment by ChihuahuaChain** [comment]
> @p0mvn tried in [rpc] and [api] , but not works

Okay I fixed adding these lines exactly on the [grpc] section, it worked.

**Comment by ValarDragon** [comment]
So whats happening if these configs aren't sent? Is the default assumed to be 0 at the moment, or something more reasonable?

**Comment by p0mvn** [comment]
The default is set to the following:
https://github.com/osmosis-labs/cosmos-sdk/blob/18b0d2de413cdce6d3307588904dcdb2156f1eb9/server/config/config.go#L238-L239

That default is only used to create the config. If someone had an old config, 0 would be assumed, and yes Bez guarded against that case here: https://github.com/osmosis-labs/cosmos-sdk/pull/220

**Comment by liangping** [comment]
works now. @p0mvn

**Comment by liangping** [comment]
> Are folks having issues on all `v7.x` versions, or just `v7.3.0`?

On both 7.2.1 and 7.3.0
